### ISSUE TITLE
Lolin s2 mini support

### ISSUE THREAD

**Issue by marcobrianza** [Aircoookie/WLED-WebInstaller#20]
I cannot flash wled to a wemos S2 mini with the web installer (esp8266 and esp32 are ok)
I have also tried the alternative bootloader without success.
I have been holding the board in bootloader mode pressing the boot button during reset.

I have also tried flashing the firmware with esptool without success for the esp32 s2 model,
the only instructions I could find apply to the esp32 (non s2).

The only way I could load the firmware is with this convoluted procedure
https://github.com/Aircoookie/WLED/issues/3706#issuecomment-1944353133

Anyway having flashed with this method I cannot upgrade wled from the web interface as would normally do because the update fails. (so I would need to use the tasmota trick again)

best regards
Marco

**Comment by blazoncek** [comment]
I have 2 Lolin Wemos S2 mini in regular use now (and several test units), updated regularly via OTA and never encountered an issue.

Truth is I flashed them 1st using PlatformIO with regular environment for Lolin S2 mini.

**Comment by marcobrianza** [comment]
> Truth is I flashed them 1st using PlatformIO with regular environment for Lolin S2 mini.
I guess this is the difference that enables the upgrade from the web installer and does not work with a vanilla board.

Are you able to load the firmware with the web installer from a board forced in bootloader mode?

**Comment by marcobrianza** [comment]
> That may be a web installer issue (open a new issue there).
this is exactly where I have opened the issue

> use esptool 
I don't find the instructions for ESP32-S2 (just ESP32)

>The easiest way is to use PlatformIO
I just had success in uploading from platform io, but I would not consider this as easy as the web installer for the majority of the users.

> As far as WLED goes, it is working correctly on Lolin S2 mini.
this is great!, I will test it in the next days

unfortunately it is just for the brave who use PlatformIO

**Comment by blazoncek** [comment]
Sorry about my mistake. I was not paying attention where the issue was opened (have too many notifications).

The instructions for esptool are the the same for all variants of ESP32. It is just files themselves that are different.

**Comment by marcobrianza** [comment]
is this bootloader also valid for esp32-s2 or is there another file?
https://github.com/Aircoookie/WLED/releases/download/v0.13.1/esp32_bootloader_v4.bin

**Comment by blazoncek** [comment]
No. S2 has its own bootloader. You may need to compile one for yourself as S2 (and other variants) may need specific bootloader depending on development board configuration.
You can try the one from [web installer](https://github.com/Aircoookie/WLED-WebInstaller/tree/master/bin/beta_0_14_2-b1).
You will also need partition map, that one depends on flash memory configuration.

**Comment by marcobrianza** [comment]
thank you for the clarification, the process is getting so complicated that flashing from platformIO is simpler until the webinstaller is fixed

**Comment by blazoncek** [comment]
FYI: https://wled.discourse.group/t/esp32-s2-wont-flash/11132/8?u=blazoncek

As of right now S2, S3 and C3 variants of ESP32 are still in *experimental* phase and not officially supported.

**Comment by Garywoo** [comment]
According to [this poster](https://github.com/esphome/esp-web-tools/issues/374#issuecomment-1941925951), release 10.0.1 of ESP Web Tools has improved the compatibility of ESP32-S2, C3, S3. 

I've tested the Tasmota implementation (although I'm not sure which release this uses), and this works and flashes my Lolin S2 Mini board.

Currently WLED Web Installer is using release 9.3.0.

**Comment by blazoncek** [comment]
@Garywoo you are welcome to make a PR if you know how to fix the installer.

**Comment by blazoncek** [comment]
For further discussion and updates, please, use https://github.com/Aircoookie/WLED-WebInstaller/issues/22#issuecomment-2051136225
